firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarAction.swift

```diff
@@ -10,22 +10,28 @@ class ToolbarAction: Action {
     let addressToolbarModel: AddressToolbarModel?
     let navigationToolbarModel: NavigationToolbarModel?
     let numberOfTabs: Int?
+    let isButtonEnabled: Bool?
 
     init(addressToolbarModel: AddressToolbarModel? = nil,
          navigationToolbarModel: NavigationToolbarModel? = nil,
          numberOfTabs: Int? = nil,
+         isButtonEnabled: Bool? = nil,
          windowUUID: WindowUUID,
          actionType: ActionType) {
         self.addressToolbarModel = addressToolbarModel
         self.navigationToolbarModel = navigationToolbarModel
         self.numberOfTabs = numberOfTabs
+        self.isButtonEnabled = isButtonEnabled
         super.init(windowUUID: windowUUID, actionType: actionType)
     }
 }
 
 enum ToolbarActionType: ActionType {
     case didLoadToolbars
+    case didLoadWebview
     case numberOfTabsChanged
+    case backButtonStateChanged
+    case forwardButtonStateChanged
```

Comment:
from our last convo, did we want to replace this with the `didLoadWebView`?

Reply:
That was my plan originally but when I looked into it the buttons were also changes when the trait collection changed so the name action name didn't fit. Do you think we should add two actions in this case? Or we could keep the actions like we have them right now and maybe improve in the future.

Reply:
Improving for the future works and we can revisit.